File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.go
func gcdOfStrings(str1 string, str2 string) string {
    if str1 == str2 {
        return str1
    }

    if str1 == "" || str2 == "" {
        return ""
    }

    if len(str2) > len(str1) {
        return gcdOfStrings(str2, str1)
    }

    return gcdOfStrings(str2, str1[len(str2):])
}

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.py
class Solution(object):
    def gcdOfStrings(self, str1, str2):
        """
        :type str1: str
        :type str2: str
        :rtype: str
        """
        if str1 == str2:
            return str1

        if len(str1) < len(str2):
            return self.gcdOfStrings(str2, str1)
        
        if str1.startswith(str2):
            return self.gcdOfStrings(str2, str1[len(str2):])

        return ""
